service/application/src/main/java/com/solace/maas/ep/event/management/agent/cli/EmaCommandLine.java

```diff
@@ -0,0 +1,210 @@
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Component
+@Order(1)
+@Slf4j
+public class EmaCommandLine implements CommandLineRunner {
+    private final ScanManager scanManager;
+    private final ImportService importService;
+    private final ScanStatusService scanStatusService;
+    private final IDGenerator idGenerator;
+    private final DataCollectionFileService dataCollectionFileService;
+    private final MessagingServiceDelegateServiceImpl messagingServiceDelegateService;
+
+    public EmaCommandLine(ScanManager scanManager,
+                          ImportService importService,
+                          ScanStatusService scanStatusService,
+                          IDGenerator idGenerator,
+                          DataCollectionFileService dataCollectionFileService,
+                          MessagingServiceDelegateServiceImpl messagingServiceDelegateService) {
+        this.scanManager = scanManager;
+        this.importService = importService;
+        this.scanStatusService = scanStatusService;
+        this.idGenerator = idGenerator;
+        this.dataCollectionFileService = dataCollectionFileService;
+        this.messagingServiceDelegateService = messagingServiceDelegateService;
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        if (args.length > 2) {
+
+            String type = args[0];
+
+            if ("scan".equals(type)) {
+                String messagingServiceId = args[1];
+                String filePathAndName = args[2];
+
+                runScan(messagingServiceId, filePathAndName);
+            } else {
+                log.error("Unknown command: {}", type);
+            }
+        } else {
+            log.info(" arguments passed to the application.");
+        }
+    }
+
+    private void runScan(String messagingServiceId, String filePathAndName) throws InterruptedException, IOException {
+        MessagingServiceEntity messagingServiceEntity = messagingServiceDelegateService.getMessagingServiceById(messagingServiceId);
+
+        ScanRequestBO scanRequestBO = new ScanRequestBO();
+        scanRequestBO.setMessagingServiceId(messagingServiceId);
+        scanRequestBO.setScanId(idGenerator.generateRandomUniqueId());
+        setScanType(messagingServiceEntity, scanRequestBO, messagingServiceId);
+        scanRequestBO.setDestinations(List.of("FILE_WRITER"));
+
+        log.info("Scan request [{}]: Received, request details: {}", scanRequestBO.getScanId(), scanRequestBO);
+        String scanId = scanManager.scan(scanRequestBO);
+
+        log.info("Scan request [{}]: Scan started.", scanId);
+        waitForScanToComplete(scanId);
+
+        if (isCompletedSuccessfully(scanId)) {
+            writeScanToZipFile(filePathAndName, scanId);
+        } else {
+            log.error("Scan request [{}]: Scan did not complete successfully.", scanId);
+        }
+    }
+
```

Comment:
It completed successfully if there are dataCollectionFileEntitites for each scan type, and none of them are marked as "FAILED"